warehouse/query-core/src/main/java/datawave/query/index/lookup/IndexInfo.java

```diff
@@ -339,203 +346,72 @@ public boolean intersect(List<JexlNode> delayedNodes) {
-                        if (null != newNode)
-                            ids.put(match.uid, newNode);
-                    }
-                    
-                    JexlNodeSet ourDelayedNodes = new JexlNodeSet();
-                    ourDelayedNodes.addAll(delayedNodes);
-                    // we may actually have no node on o
-                    if (null != o.getNode())
-                        ourDelayedNodes.add(o.getNode());
-                    
-                    Set<IndexMatch> matches = buildNodeList(ids, IndexMatchType.AND, true, Lists.newArrayList(ourDelayedNodes.getNodes()));
-                    
-                    merged.uids = ImmutableSortedSet.copyOf(matches);
-                    merged.count = merged.uids.size();
-                } else if (o.onlyEvents()) {
-                    /*
-                     * E) We have LARGE AND SMALL
-                     */
-                    HashMultimap<String,JexlNode> ids = HashMultimap.create();
-                    for (IndexMatch match : o.uids) {
-                        JexlNode newNode = match.getNode();
-                        if (null != newNode)
-                            ids.put(match.uid, newNode);
-                    }
-                    
-                    JexlNodeSet ourDelayedNodes = new JexlNodeSet();
-                    ourDelayedNodes.addAll(delayedNodes);
-                    // possible, depending on how query is processed that we have no node.
-                    if (null != getNode())
-                        ourDelayedNodes.add(getNode());
-                    
-                    Set<IndexMatch> matches = buildNodeList(ids, IndexMatchType.AND, true, Lists.newArrayList(ourDelayedNodes.getNodes()));
-                    merged.uids = ImmutableSortedSet.copyOf(matches);
-                    merged.count = merged.uids.size();
-                } else {
-                    
-                    merged.count = Math.min(count, o.count);
-                    merged.uids = ImmutableSortedSet.of();
-                }
-            }
+            // else we are left countably infinite shard ranges. Take the minimum count.
+            merged.count = Math.min(count, o.count);
+            merged.uids = ImmutableSortedSet.of();
         }
-        JexlNodeSet nodeSet = new JexlNodeSet();
-        if (null != myNode && myNode != o.myNode)
-            nodeSet.add(myNode);
-        if (null != o.myNode)
-            nodeSet.add(o.myNode);
-        nodeSet.addAll(delayedNodes);
-        merged.myNode = TreeFlatteningRebuildingVisitor.flatten(JexlNodeFactory.createAndNode(nodeSet.getNodes()));
+        
+        // now handle updating the top level node
+        JexlNodeSet nodes = new JexlNodeSet();
+        if (this.getNode() != null) {
+            nodes.add(this.getNode());
+        }
+        if (o.getNode() != null) {
+            nodes.add(o.getNode());
+        }
+        if (!delayedNodes.isEmpty()) {
```

Comment:
The check for !delayedNodes.isEmpty() is redundant.  addAll already does that.

Reply:
Added